src/relentless/model/potential/dihedral.py

```diff
@@ -0,0 +1,482 @@
+            u = u.item()
+        return u
+
+    def force(self, type_, phi):
+        """Evaluate dihedral force."""
+        params = self.coeff.evaluate(type_)
+        c1 = params["c1"]
+        c2 = params["c2"]
+        c3 = params["c3"]
+        c4 = params["c4"]
+
+        phi, f, s = self._zeros(phi)
+
+        psi = phi - numpy.pi
+
+        f = (
+            c1 * numpy.sin(psi)
+            + 2 * c2 * numpy.cos(psi) * numpy.sin(psi)
+            + 3 * c3 * numpy.cos(psi) ** 2 * numpy.sin(psi)
+            + 4 * c4 * numpy.cos(psi) ** 3 * numpy.sin(psi)
+        )
+        if s:
+            f = f.item()
+        return f
+
+    def _derivative(self, param, phi, c0, c1, c2, c3, c4, c5):
+        r"""Evaluate dihedral derivative with respect to a variable."""
+        phi, d, s = self._zeros(phi)
+
+        psi = phi - numpy.pi
+
+        if param == "c0":
+            d = numpy.ones_like(psi)
+        elif param == "c1":
+            d = numpy.cos(psi)
+        elif param == "c2":
+            d = numpy.cos(psi) ** 2
+        elif param == "c3":
+            d = numpy.cos(psi) ** 3
+        elif param == "c4":
+            d = numpy.cos(psi) ** 4
+        elif param == "c5":
+            d = numpy.cos(psi) ** 5
+        else:
+            raise ValueError("Unknown parameter")
+
+        if s:
+            d = d.item()
+        return d
+
+
+class DihedralSpline(potential.BondedSpline, DihedralPotential):
+    """Spline dihedral potentials.
+
+    The dihedral spline potential is defined by interpolation through a set of
+    knot points. The interpolation scheme uses Akima splines.
+
+    Parameters
+    ----------
+    types : tuple[str]
+        Types.
+    num_knots : int
+        Number of knots.
+    mode : str
+        Mode for storing the values of the knots in
+        :class:`~relentless.variable.Variable` that can be optimized. If
+        ``mode='value'``, the knot amplitudes are manipulated directly.
+        If ``mode='diff'``, the amplitude of the *last* knot is fixed, and
+        differences between neighboring knots are manipulated for all other
+        knots. Defaults to ``'diff'``.
+    name : str
+        Unique name of the potential. Defaults to ``__u[id]``, where ``id`` is the
+        unique integer ID of the potential.
+
+    Examples
+    --------
+    The spline potential is setup from a tabulated potential instead of
+    specifying knot parameters directly::
+
+        spline = relentless.potential.dihedral.DihedralSpline(
+            types=("dihedralA",),
+            num_knots=3,
+        )
+        spline.from_array(("dihedralA"),[0,1,2],[4,2,0])
+
+    However, the knot variables can be iterated over and manipulated directly::
+
+        for r,k in spline.knots("dihedralA"):
+            k.value = 1.0
+
+    """
+
+    _space_coord_name = "phi"
+
+    def __init__(self, types, num_knots, mode="diff", name=None):
+        super().__init__(types=types, num_knots=num_knots, mode=mode, name=name)
```

Comment:
I'm not sure if I missed this in prior PRs, but if all of this is directly passed through I think there is no need to write this function out explicitly. (It will inherit.)